pkg/virtual/framework/forwardingregistry/store.go

```diff
@@ -176,10 +177,20 @@ func DefaultDynamicDelegatedStoreFuncs(
 			return nil, false, err
 		}
 
+		requestInfo, _ := genericapirequest.RequestInfoFrom(ctx)
+
 		doUpdate := func() (*unstructured.Unstructured, error) {
 			oldObj, err := s.Get(ctx, name, &metav1.GetOptions{})
 			if err != nil {
-				return nil, err
+				// Continue on 404 when forceAllowCreate is enabled,
+				// or for PATCH requests, so server-side apply requests
+				// for non-existent objects can still be processed.
+				if !apierrors.IsNotFound(err) ||
+					!forceAllowCreate &&
+						!(requestInfo != nil && requestInfo.Verb == "patch") {
```

Comment:
What about `patch` requests which are not SSA requests ?
Will they try to create it as well ?

Reply:
For JSON or SMP patch requests, the corresponding patcher implementation returns a `errors.NewNotFound` error, so the response code is 404 as expected. The patchers execute as part of the `objInfo.UpdatedObject(ctx, oldObj)` call, and their implementations can be found at https://github.com/kubernetes/apiserver/blob/48405445f09218bb8b7a8a81eb21027cfd65761b/pkg/endpoints/handlers/patch.go, with the errors at:
- https://github.com/kubernetes/apiserver/blob/48405445f09218bb8b7a8a81eb21027cfd65761b/pkg/endpoints/handlers/patch.go#L355
- https://github.com/kubernetes/apiserver/blob/48405445f09218bb8b7a8a81eb21027cfd65761b/pkg/endpoints/handlers/patch.go#L451

Reply:
Thanks for the explanation.
Do you think you could add a comment in the code to explain this ? It's not easy to think of or guess when reading.

Reply:
Sure, I've added eb9e64dba2c064ebce3578c08ad8c2105ef002d3 that hopefully covers it.

I've also pushed 6d1b96804c3b9efb1a69cc49597fb6cfba6b4647 to have the use case covered in e2e tests.